fluent/fluent.go

```diff
@@ -347,13 +353,25 @@ func (f *Fluent) connect() (err error) {
 }
 
 func (f *Fluent) run() {
+	drainEvents := false
 	for {
+		select {
+		case stopRunning, ok := <-f.stopRunning:
+			if stopRunning || !ok {
+				drainEvents = true
+			}
+default:
```

Comment:
Could you use non-`default` label here (and specify it on `continue`)?
This is a part of the entire loop, and going back to here is a kind of exceptional case. So it should be not `default`.